scio-core/src/main/scala/com/spotify/scio/values/SideSet.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2019 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.scio.values
+
+import scala.collection.mutable.{ArrayBuffer, Map => MMap}
+
+case class SideSet[K](side: SideInput[MMap[K, ArrayBuffer[Unit]]])
```

Comment:
We might use something like this have a separate function `combineAsSetSideInput`
```suggestion
case class SideSet[K](side: SideInput[Set[K]])
```
Just thinking.

Reply:
That would work too, but would mean more code - by defining Set in terms of Map some things become simpler.
This function in particular becomes simple:
`def toSideSet(implicit coder: Coder[T]): SideSet[T] = SideSet(self.map((_, ())).toSideMap.side)`